opengrok-indexer/src/main/java/org/opengrok/indexer/history/Annotation.java

```diff
@@ -190,6 +191,53 @@ public int getFileVersionsCount() {
         return fileVersions.size();
     }
 
+    /**
+     * Generate the color palette for the annotated revisions.
+     * <p>
+     * First, take into account revisions which are tracked in history fields
+     * and compute their color. Secondly, use all other revisions in order
+     * which is undefined and generate the rest of the colors for them.
+     *
+     * @return map of (revision, css color string) for each revision in {@code getRevisions()}
+     * @see #getRevisions()
+     */
+    public Map<String, String> getColors() {
+        Map<String, String> colors = new HashMap<>();
+        boolean even = true;
+        int hue = 60;
```

Comment:
I'd assume the most recent version would start with red and not yellow.

Reply:
It used to be yellow so I kept it consistent.

Reply:
Maybe it's time to change it to red :-) The old and new schemes are so different it does not matter with what color to start. Or maybe better yet: make it consistent with IDEA behavior.

Reply:
After brief overview IDEA uses fixed linear gradient for 5 stop colors (roughly green, gray, blue, yellow, red) with interleaving 4 steps. Generating total 21 color and distribute the revisions among this pallete. I can prepare an example for that although I find this hsl approach nicer.

Reply:
<img width="516" alt="Screenshot 2019-04-06 at 11 24 50" src="https://user-images.githubusercontent.com/6997160/55667611-aaf71380-585e-11e9-9a2c-8352609b3d0e.png">

Reply:
I like the HSL approach better, partly also because the saturation is higher. On the other hand, lots of folks are used to IDEA these days and could be confused with our custom scheme.

What happens in IDEA if there are more than 21 revisions ? Do these work like hash buckets ?

Reply:
I know the picture here looks little bit awful but in reality it looks quite ok on the opengrok page.

For 22+ revisions this is like a hash buckets yes. The interval of revisions is mapped to the single target color.